SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java

```diff
@@ -113,6 +113,7 @@ private void submitRequest(SingularityRequest request, Optional<SingularityReque
 
     if (oldRequest.isPresent()) {
       authorizationHelper.checkForAuthorization(oldRequest.get(), user, SingularityAuthorizationScope.WRITE);
+      authorizationHelper.checkForAuthorizedChanges(request, oldRequest.get(), user.get()); // previous check guarantees that user is present
```

Comment:
comment isn't exactly right -- if auth is disabled it's not going to check if user is present or not (https://github.com/HubSpot/Singularity/blob/4b3e6f90860bf65734ed1ffef0dee7241093b425/SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityAuthorizationHelper.java#L136)